pkg/deployment/nova.go

```diff
@@ -0,0 +1,137 @@
+/*
+Copyright 2023.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package deployment
+
+import (
+	"context"
+	"fmt"
+
+	appsv1 "k8s.io/api/apps/v1"
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+
+	dataplanev1beta1 "github.com/openstack-k8s-operators/dataplane-operator/api/v1beta1"
+	condition "github.com/openstack-k8s-operators/lib-common/modules/common/condition"
+	"github.com/openstack-k8s-operators/lib-common/modules/common/helper"
+	novav1beta1 "github.com/openstack-k8s-operators/nova-operator/api/v1beta1"
+)
+
+// DeployNovaCompute deploys the nova compute configuration and services
+func DeployNovaExternalCompute(ctx context.Context, helper *helper.Helper, obj client.Object, sshKeySecret string, inventoryConfigMap string, status *dataplanev1beta1.OpenStackDataPlaneNodeStatus, networkAttachments []string, openStackAnsibleEERunnerImage string) (ctrl.Result, error) {
+
+	log := helper.GetLogger()
+
+	novaExternalCompute, err := GetNovaExternalCompute(ctx, helper, obj, "openstackdataplanenode")
```

Comment:
I think you don't need to Get the NovaExternalComptue to be able to use controllerutil.CreateOrPatch. The `novaExternalCompute` object you pass to CreateOrPatch only need to have the ObjectMeta part filled so you don't have to read the the whole object first from the API.

for example https://github.com/openstack-k8s-operators/nova-operator/blob/452df1184a022e1f33291463251aeb08160a65ef/controllers/nova_controller.go#L622-L660